Packs/DomainTools_Iris/Scripts/DomainToolsSetIndicatorTable/set_indicator_table_data.py

```diff
@@ -0,0 +1,86 @@
+
+
+def find_indicator_reputation(domain_age, proximity_score, threat_profile_score):
+    proximity_score_threshold = int(demisto.args().get('proximity_score_threshold', 70))
+    age_threshold = int(demisto.args().get('age_threshold', 7))
+    threat_profile_score_threshold = int(demisto.args().get('threat_profile_score_threshold', 70))
+
+    if proximity_score > proximity_score_threshold or threat_profile_score > threat_profile_score_threshold:
+        return 'Bad'
+    elif domain_age < age_threshold and (
+            proximity_score < proximity_score_threshold or threat_profile_score < threat_profile_score_threshold):
+        return 'Suspicious'
+    else:
+        return 'Good'
+
+
+def main():
+    domaintools_data = demisto.args().get("domaintools_data", None)
+
+    human_readable_str = "No context data for domain."
+    if domaintools_data:
+        domain_name = domaintools_data.get("Name")
+        domaintools_hosting_data = domaintools_data.get("Hosting", {})
+        domaintools_identity_data = domaintools_data.get("Identity", {})
+        domaintools_analytics_data = domaintools_data.get("Analytics", {})
+
+        create_date = domaintools_data.get("Registration", {}).get("CreateDate")
+        domain_age = 0
+        if create_date:
+            domain_age = find_age(create_date)
+
+        try:
+            threat_profile_score = domaintools_analytics_data.get(
+                "ThreatProfileRiskScore", {}
+            ).get("RiskScore", 0)
+            proximity_risk_score = domaintools_analytics_data.get(
+                "ProximityRiskScore", 0
+            )
+            reputation = find_indicator_reputation(domain_age, proximity_risk_score, threat_profile_score)
+        except Exception:
+            reputation = "Unknown"
+
+        demisto_indicator = {
+            "type": "Domain",
+            "value": domain_name,
+            "source": "DomainTools",
+            "reputation": reputation,
+            "seenNow": "true",
+            "ipaddresses": json.dumps(domaintools_hosting_data.get("IPAddresses")),
+            "ipcountrycode": domaintools_hosting_data.get("IPCountryCode"),
+            "mailservers": json.dumps(domaintools_hosting_data.get("MailServers")),
+            "spfrecord": domaintools_hosting_data.get("SPFRecord"),
+            "nameservers": domaintools_hosting_data.get("NameServers"),
+            "sslcertificate": json.dumps(
+                domaintools_hosting_data.get("SSLCertificate")
+            ),
+            "soaemail": domaintools_identity_data.get("SOAEmail"),
+            "sslcertificateemail": domaintools_identity_data.get("SSLCertificateEmail"),
+            "emaildomains": domaintools_identity_data.get("EmailDomains"),
+            "additionalwhoisemails": domaintools_identity_data.get(
+                "AdditionalWhoisEmails"
+            ),
+            "domainage": domain_age,
+        }
+        demisto.executeCommand("createNewIndicator", demisto_indicator)
```

Comment:
Use the `Common.Domain` object from `CommonServerPython` for creating an indicator.

Reply:
Just wanted to clarify here. Did you mean for us to continue using the demisto.executeCommand("createNewIndicator", indicator) call, but pass in a Common.Domain object instead of a python dict? Or did you mean for us to use a different method for creating the new indicator? We didn't see a way to save the Common.Domain indicator once it was created.

Here's a snippet we are using 
` return CommandResults(readable_output=human_readable_str, indicator=domain_indicator)`

checking it has an output however, it doesn't create an indicator.

Reply:
This is probably because you're passing to the `indicator` parameter a dict instead of an indicator object (in this case, I believe it should be a `Common.Domain` object?).

You can see an example for a proper `CommandResults` object with an indicator [here](https://github.com/demisto/content/blob/master/Packs/CommonScripts/Scripts/ResolveShortenedURL/ResolveShortenedURL.py#L356-L369).

Reply:
sorry for the confusion, we are passing a `Common.Domain` object to command result. 
here's the `domain_indicator` value.
 `domain_indicator = Common.Domain(domain=domain_name,
                                         creation_date=create_date,
                                         dbot_score=dbot_score
                                         )`

Reply:
I'm not sure indicators should appear in the debug panel there.
They should be added to XSOAR's "Threat Intel" panel.

Passing a custom dict like that is non-conventional and can cause all-sort of unexpected bugs & issues, since there is some additional processing done on indicator objects on the `CommonServerPython` module (when running `return_Results` with a `CommandResults` object).

Reply:
Thanks @MichaelYochpaz, is it good idea if we just create our own/custom indicator type to cater the fields we are defining when creating an indicator? 

we are adding this custom fields:
<meta charset="utf-8"><b style="font-weight:normal;" id="docs-internal-guid-d83d8736-7fff-7645-ebaf-abaea1e8bde9"><div dir="ltr" style="margin-left:36pt;" align="left">

additionalWhoisEmails
--
domainAge
emailDomains
ipAddresses
mailServers
nameServers
soaEmail
spfRecord
sslCertificate

</div></b>

Reply:
As per discussed internally, we will move forward on creating a custom indicator type e.g. (Domaintools Iris) and add the custom fields.